src/eve/concepts.py

```diff
@@ -170,7 +177,13 @@ def __new__(mcs, name, bases, namespace, **kwargs):
         return cls
 
 
-class BaseNode(BaseModel, metaclass=NodeMetaclass):
+AnyNode = TypeVar("AnyNode", bound="BaseNode")
```

Comment:
This node has already been there before, but the naming seems confusing since the standard types, e.g. `bool`, `int` etc., which are valid nodes themselves do not match. Further there are no other `BaseNode` like classes just using BaseNode in the annotations should be fine.

Reply:
It is slightly different to use BaseNode or AnyNode in the annotations. BaseNode implies you are using only features of the BaseNode class (which of course are also implemented by any of its subclasses) and thus a signature like `Callable[[BaseNode], BaseNode]` can accept a BaseNode subclass A and return a different subclass B. AnyNode is a TypeVar and thus it allows us to define type uses with more detail. The previous example using AnyNode  `Callable[[AnyNode], AnyNode]` means that a function accepts BaseNode subclasses but returns the same type it gets.